src/utils/useStateDecodeParams/index.ts

```diff
@@ -0,0 +1,56 @@
+import useSWR from 'swr'
+import LotusRpcEngine from '@glif/filecoin-rpc-client'
+import { validateAddressString } from '@glif/filecoin-address'
+
+import { useEnvironment } from '../../services/EnvironmentProvider'
+
+interface UseStateDecodeParamsResult<T> {
+  params: T | null
+  loading: boolean
+  error: Error | null
+}
+
+const fetcher = async <T>(
+  lotusApi: LotusRpcEngine,
+  lotusMethod: string,
+  actorAddress: string,
+  methodNum: number,
+  base64Params: string
+): Promise<T | null> => {
+  if (!actorAddress || !base64Params) return null
+
+  if (!validateAddressString(actorAddress))
+    throw new Error('Invalid actor address')
+
+  // Temporarily attempt to parse as JSON string,
+  // until StateDecodeParams is removed from graph
+  try {
+    return JSON.parse(base64Params)
+  } catch {
+    // params are not a JSON string
+  }
+
+  return await lotusApi.request<T>(
```

Comment:
Eventually i think we should move this call to the graph and cache it since i imagine we'll be making the same call over and over

Reply:
Yeah I think we should move it to the graph as well but then we don't need to call 'StateDecodeParams' from Lotus, we can use the GoLang types to unmarshall CBOR and marshall to JSON. We should do this when we circle back to this scope of work.